app/models/manager_refresh/inventory_collection.rb

```diff
@@ -517,17 +527,19 @@ def object_index_with_keys(keys, record)
     def noop?
       # If this InventoryCollection doesn't do anything. it can easily happen for targeted/batched strategies.
       if targeted?
-        if parent_inventory_collections.nil? && manager_uuids.blank? && skeletal_manager_uuids.blank? &&
-           all_manager_uuids.nil? && parent_inventory_collections.blank? && custom_save_block.nil?
+        if parent_inventory_collections.nil? && manager_uuids.blank? &&
+           all_manager_uuids.nil? && parent_inventory_collections.blank? && custom_save_block.nil? &&
+           skeletal_primary_index.blank?
```

Comment:
This looks like it can use a helper method, `inventory_blank?` or something

Reply:
Right, I have a noop? refactoring in followup PR